### PR TITLE
re-write components without styled-components

### LINKED ISSUE #1041: Re-write Flex without styled-components

### PR DESCRIPTION
**Pull request by jasonbasuil**
closes #1041

Removing `styled-components` as there are some compatibility issues when using SSR on RS. This just re-writes a couple components to use CSS modules instead.

Chromatic: https://62d040e741710e4f085e0647-wmozfzrxln.chromatic.com/?path=/story/layouts-flex--default

### CODE REVIEW

File: src/Flex/Flex.module.scss [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  display: flex;
+}
+
+.align-items-stretch {
+  align-items: stretch;
+}
+
+.align-items-center {
+  align-items: center;
+}
```
Comment (jasonbasuil) [inline src/Flex/Flex.module.scss:15]: I wanted to stick to the convention of using pascalCase with CSS modules here, but there were so many properties and it was a bit of an eye sore to parse and it seemed easier to read with Kebabcase

File: src/Flex/Flex.stories.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ export const Default: Story = {
     flexDirection: 'row',
     justifyContent: 'space-between',
   },
-  render: ({ children, ...args }) => (
```
Comment (jasonbasuil) [inline src/Flex/Flex.stories.tsx:28]: gets rid of the unused children warning

File: src/Popover/PopoverCard.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  children?: ReactNode;
+  size?: 'sm' | 'md';
+}
+
+const PopoverCard = ({
+  children,
+  className,
+  size = 'sm',
+}: PopoverCardProps) => {
+  const widthClass = handleWidth(size);
```
Comment (jasonbasuil) [inline src/Popover/PopoverCard.tsx:27]: we use PopoverCard once in RS. Verified it still works fine:

https://github.com/user-interviews/ui-design-system/assets/37383785/ecb5e6cd-7605-4a88-b9da-9f2eff80df78

File: src/Flex/Flex.tsx
Comment (kyleshike) [inline src/Flex/Flex.tsx:1]: I feel like we should revisit this component at some point.  Its a very thin wrapper over flexbox styles. I'm not sure that it really abstracts anything or adds any guardrails, etc.

**Review by kyleshike — APPROVED** [review]
Looks great.  Left a note on the flex component, but not blocking.

File: src/Popover/PopoverCard.tsx (hunk 3 above)
Comment (careful-axe) [reply src/Popover/PopoverCard.tsx:27]: I think we might want to use this for segments soon as well.

File: src/Flex/Flex.tsx
Comment (jasonbasuil) [reply src/Flex/Flex.tsx:1]: Yeah it is pretty simple, but if I can center basic stuff without more CSS that's a win.  Can revisit to make it more robust down the road.

**Review by careful-axe — APPROVED** [review]
Thanks for keeping me in the loop!

**Pull request merged**